internal/services/eventhub/eventhub_namespace_resource_test.go

```diff
@@ -399,7 +400,7 @@ func TestAccEventHubNamespace_BasicWithSkuUpdate(t *testing.T) {
 			),
 		},
 		{
-			Config: r.premium(data),
+			Config: r.premium(data, features.FourPointOhBeta()),
```

Comment:
I appreciate the effort here to reduce the lines of code and repetition, but keeping the clean up effort of all the `FourPointOh`/`FourPointOhBeta` code in mind, this is more work to unpick than if the config is just duplicated.

We try and leave the test alone if possible and update the function returning the config to return the right config based on the flag, so in this case

```
func (EventHubNamespaceResource) premium(data acceptance.TestData) string {
	if !features.FourPointOhBeta() {
		return fmt.Sprintf(`
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-eh-%d"
  location = "%s"
}

resource "azurerm_eventhub_namespace" "test" {
  name                = "acctesteventhubnamespace-%d"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku                 = "Premium"
  capacity            = "1"
  zone_redundant      = true
}
`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
	}
	
	return fmt.Sprintf(`
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-eh-%d"
  location = "%s"
}

resource "azurerm_eventhub_namespace" "test" {
  name                = "acctesteventhubnamespace-%d"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku                 = "Premium"
  capacity            = "1"
}
`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
}
```

Reply:
Fixed.